globus_cli/commands/login.py

```diff
@@ -97,7 +115,50 @@ def do_login_flow():
     auth_code = click.prompt(
         'Enter the resulting Authorization Code here').strip()
 
-    # exchange, done!
+    # finish login flow
+    exchange_code_and_store_config(native_client, auth_code)
+
+
+def do_local_server_login_flow():
+    """
+    Starts a local http server, opens a browser to have the user login,
+    and gets the code redirected to the server (no copy and pasting required)
+    """
+    # start local server and create matching redirect_uri
+    for port in range(8000, 8999):
```

Comment:
I think instead of trying to play this game we should let the OS give us a port at runtime. Something like:

```
server = start_local_server(listen=('127.0.0.1', 0))
_, port = server.socket.getsockname()
redirect_uri = 'http://localhost:{}'.format(port)
```

Passing in `0` tells the underlying socket builder to just ask the OS for a free port.